pkg/controller/master/supportbundle/controller.go

```diff
@@ -30,37 +34,80 @@ func (h *Handler) OnSupportBundleChanged(key string, sb *harvesterv1.SupportBund
 	}
 
 	switch sb.Status.State {
-	case StateNone:
+	case types.StateNone:
 		logrus.Debugf("[%s] generating a support bundle", sb.Name)
 		err := h.manager.Create(sb, settings.SupportBundleImage.Get())
-		toUpdate := sb.DeepCopy()
 		if err != nil {
-			h.setError(toUpdate, fmt.Sprintf("fail to create manager for %s: %s", sb.Name, err))
-		} else {
-			h.setState(toUpdate, StateGenerating)
+			return h.setError(sb, fmt.Sprintf("fail to create manager for %s: %s", sb.Name, err))
 		}
-		return h.supportBundles.Update(toUpdate)
+		return h.setState(sb, types.StateGenerating)
+	case types.StateGenerating:
+		logrus.Debugf("[%s] support bundle is being generated", sb.Name)
+		return h.checkManagerStatus(sb)
 	default:
 		logrus.Debugf("[%s] noop for state %s", sb.Name, sb.Status.State)
 		return sb, nil
 	}
 }
 
-func (h *Handler) setError(toUpdate *harvesterv1.SupportBundle, reason string) {
-	logrus.Errorf(reason)
-	harvesterv1.SupportBundleInitialized.False(toUpdate)
-	harvesterv1.SupportBundleInitialized.Message(toUpdate, reason)
+func (h *Handler) checkManagerStatus(sb *harvesterv1.SupportBundle) (*harvesterv1.SupportBundle, error) {
+	if time.Now().UTC().After(sb.CreationTimestamp.Add(types.BundleCreationTimeout)) {
```

Comment:
I think we don't need to call UTC() for the comparison

Reply:
[Fixed](https://github.com/harvester/harvester/pull/844/commits/4d0240934fdb0f340e98b0d5afc70d13e9a1dd46)